### PR TITLE
Add required dependency between Crashlytics and Session

### PR DESCRIPTION
**Pull request by samedson**
### History
Alternative to https://github.com/firebase/firebase-ios-sdk/pull/10619 where the dependency is required.
Replaces https://github.com/firebase/firebase-ios-sdk/pull/10636

### TODO
- [x] Add tests done in https://github.com/firebase/firebase-ios-sdk/pull/10669
- [x] Ensure future backgrounds still resolve the promises done in https://github.com/firebase/firebase-ios-sdk/pull/10669
- [ ] Send the Session ID with the crash report (going to follow up)

### CODE REVIEW

File: FirebaseSessions/Sources/Public/SessionsSubscriber.swift [hunk 1]
```diff
@@ -28,7 +28,7 @@ public protocol SessionsSubscriber {
 /// whenever the Session changes
 @objc(FIRSessionDetails)
 public class SessionDetails: NSObject {
-  var sessionId: String?
+  @objc public var sessionId: String?
```
Comment (leotianlizhan) [inline FirebaseSessions/Sources/Public/SessionsSubscriber.swift:31]: FYI I moved this to a separate PR that I hope we could merge early because this fix is needed for my PR to work as well

Comment (samedson) [reply FirebaseSessions/Sources/Public/SessionsSubscriber.swift]: Awesome - thanks for doing that
